pentasharp/Controllers/UserActivityController.cs

```diff
@@ -41,5 +57,152 @@ public async Task<IActionResult> MyReservations()
                 return View("Error", new { message = "An error occurred while retrieving your reservations." });
             }
         }
-    }
-}
+
+        [HttpGet("MyBookings")]
+        public async Task<IActionResult> MyBookings()
+        {
+
+            var userId = _httpContextAccessor.HttpContext.Session.GetInt32("UserId");
+
+            try
+            {
+                var bookings = await _taxiBookingService.GetBookingsForUserAsync(userId.Value);
+                return View("MyBookings", bookings);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "An error occurred while retrieving reservations for user {UserId}", userId);
+                return View("Error", new { message = "An error occurred while retrieving your reservations." });
+            }
+        }
+
+        [HttpPost("CancelBooking/{id}")]
+        public async Task<IActionResult> CancelBooking(int id)
+        {
+            var userId = _httpContextAccessor.HttpContext.Session.GetInt32("UserId");
```

Comment:
the check if the user exists or not can be moved to service not controller, do this change throughout the controller